Allow installing CA from a url in install_ca

cescgina opened the pull request:
Allow downloading a certificate file directly into
/etc/pki/ca-trust/source/anchors/ in install_ca role, whitout it
being part of an rpm. This change also introduces new tasks to add CAs
to an openshift image registry in the openshift_setup role.

As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] README in the role
  - [x] Content of the docs/source is reflecting the changes

cjeanner commented on roles/reproducer/tasks/install_ca.yml:
Did you consider extending the install_ca role to support downloading a remote CA?
If you download the file, you'll need to run the `sudo update-ca-trust` once it's in the `anchors` directory.

cescgina commented:
@cjeanner @viroel I've updated the PR. The change to install_ca to pull from ca was simple enough, but I also introduced a new role to do the same job that was done by the downstream hook. I'm not sure I covered all cases there, please take a look when free

cescgina commented:
recheck

cjeanner commented on roles/configure_ocp_registry_ca/tasks/main.yml:
```diff
@@ -0,0 +1,39 @@
+    - name: Create config map with registry CAs
+      ansible.builtin.command:
+        cmd: >-
+          "oc create configmap registry-cas -n openshift-config --from-file={{ cifmw_configure_ocp_registry_ca_registry_to_add }}={{ cifmw_configure_ocp_registry_ca_bundle_path }}"
```
```suggestion
          oc create configmap registry-cas -n openshift-config
          --from-file={{ cifmw_configure_ocp_registry_ca_registry_to_add }}={{ cifmw_configure_ocp_registry_ca_bundle_path }}
```

would be slightly better - but the `--from-file` is long on its own ^^'

cescgina replied:
I've split the command in two lines and moved the `--from-file` to a private var so it's a bit shorter

cjeanner commented on roles/configure_ocp_registry_ca/tasks/main.yml:
```diff
@@ -0,0 +1,39 @@
+    - name: Install Red Hat CA for pulling images from internal registry
+      ansible.builtin.command:
+        cmd: >-
+          oc patch image.config.openshift.io/cluster --patch '{"spec":{"additionalTrustedCA":{"name":"registry-cas"}}}' --type=merge
```
```suggestion
          oc patch image.config.openshift.io/cluster
          --patch '{"spec":{"additionalTrustedCA":{"name":"registry-cas"}}}'
          --type=merge
```

cescgina replied:
done, thanks

cjeanner commented on roles/install_ca/tasks/main.yml:
```diff
@@ -22,6 +22,12 @@
 
+    - name: Install internal CA from url
+      when: cifmw_install_ca_url is defined
+      ansible.builtin.get_url:
```
we may want to allow to not check TLS certificate at this point?

cescgina replied:
I'm not sure what you're asking, is there some parameter to `get_url` that it's missing?

cescgina replied:
I missed that parameter, thanks @cjeanner updated the patch with a variable that controls whether we validate the certs

cjeanner commented on roles/configure_ocp_registry_ca/molecule/default/converge.yml:
```diff
@@ -0,0 +1,60 @@
+      register: _image_patch
+
+    - name: Debug cluster was patched
+      ansible.builtin.deub:
```
*debug

cescgina replied:
indeed, thanks!

cjeanner commented:
/approve

rlandy commented:
/lgtm

cescgina commented:
recheck

cescgina commented:
recheck

cescgina commented:
recheck

pablintino commented on roles/configure_ocp_registry_ca/tasks/main.yml:
```diff
@@ -0,0 +1,44 @@
+    - name: Create config map with registry CAs
+      ansible.builtin.command:
+        cmd: >-
+          oc create configmap registry-cas -n openshift-config
```
I'm ok creating a role for such of a simple task, but the main goal of extracting code in a dedicated role is to allow it to be reused. The way this is coded will prevent it's re-usability, cause 2 calls to` oc create` for the same type and name will fail.

pablintino commented on roles/configure_ocp_registry_ca/tasks/main.yml:
```diff
@@ -0,0 +1,44 @@
+          oc create configmap registry-cas -n openshift-config
+          --from-file="{{ _config_map_data }}"
+
+    - name: Install Red Hat CA for pulling images from internal registry
```
While the command used here is fine the task will be flagged as changed always. Can we make it idempotent or just skip it if the CM is already added?

pablintino reviewed (changes requested):
While I'm ok with the general approach of this and I see that it's needed, do we really need a dedicated role for this?
